server/konnectors/aprr.js

```diff
@@ -0,0 +1,202 @@
+'use strict';
+
+const request = require('request').defaults({
+  jar: true,
+  rejectUnauthorized: false,
+  followAllRedirects: true   // allow redirections
+});
+const moment = require('moment');
+const cheerio = require('cheerio');
+const baseKonnector = require('../lib/base_konnector');
+
+const filterExisting = require('../lib/filter_existing');
+const localization = require('../lib/localization_manager');
+const saveDataAndFile = require('../lib/save_data_and_file');
+const linkBankOperation = require('../lib/link_bank_operation');
+
+const log = require('printit')({
+  prefix: 'APRR',
+  date: true,
+});
+
+const fileOptions = {
+  vendor: 'APRR',
+  dateFormat: 'YYYYMMDD',
+};
+
+const Bill = require('../models/bill');
+
+// Konnector
+const connector = module.exports = baseKonnector.createNew({
+  name: 'APRR',
+  fields: {
+    login: 'text',
+    password: 'password',
+    folderPath: 'folder',
+  },
+  models: [Bill],
+  fetchOperations: [
+    getHiddenInputs,
+    logIn,
+    fetchBillingInfo,
+    parsePage,
+    customFilterExisting,
+    customSaveDataAndFile,
+    linkBankOperation({
+      log,
+      model: Bill,
+      identifier: 'APRR',
+      minDateDelta: 4,
+      maxDateDelta: 20,
+      amountDelta: 0.1,
+    }),
+    buildNotifContent,
+  ],
+});
+
+// Procedure to get hidden inputs
+function getHiddenInputs(requiredFields, bills, data, next) {
+  const url = 'https://espaceclient.aprr.fr/aprr/Pages/connexion.aspx';
+  const options = {
+    url,
+    method: 'GET',
+  };
+
+  connector.logger.info('Getting the hidden inputs...');
+
+  request(options, (err, res, body) => {
+    if (err) return next(err);
+
+    const $ = cheerio.load(body);
+
+    var obj = {};
+    $( "body" ).find("input[type='hidden']").each(function(){
+		obj[$(this).attr('name')] = $(this).val();
+	});
+
+	// adding login/pwd
+	obj['ctl00$PlaceHolderMain$TextBoxLogin'] = requiredFields.login;
+	obj['ctl00$PlaceHolderMain$TextBoxPass'] = requiredFields.password;
+	obj['ctl00$PlaceHolderMain$ImageButtonConnection.x'] = Math.floor((Math.random() * 10) + 1);
+	obj['ctl00$PlaceHolderMain$ImageButtonConnection.y'] = Math.floor((Math.random() * 10) + 1);
+
+	data.inputs = obj;
+
+    //connector.logger.info('inputs', data.inputs);
+    return next();
+  });
+};
+
+
+// Procedure to login
+function logIn(requiredFields, bills, data, next) {
+
+  const options = {
+    method: 'POST',
+    url: 'https://espaceclient.aprr.fr/aprr/Pages/connexion.aspx',
+    form: data.inputs,
+	headers: {
+        'Referer': 'https://espaceclient.aprr.fr/aprr/Pages/connexion.aspx',
+     },
+  };
+
+  connector.logger.info('Logging in on APRR website...');
+
+  request(options, (err, res, body) => {
+    if (err) return next(err);
+
```

Comment:
the wensite returns HTTP 200 even if the login failed...
I can try to match a string to determine if the login worked but seems not so good...

do you have a smart idea to check this ?